### PR TITLE
Migrate to tensorflow.js

### PR DESCRIPTION
**Pull request by amellnik**
This builds on top of #10 and #13.  

This relies on a few commits since latest tagged, but the recent shape work breaks it -- this was tested against https://github.com/FluxML/Flux.jl/commit/4320738d870bfefa07f99cbabdf9367f9e9bdb6f.

### CODE REVIEW

**Comment by MikeInnes** [comment]
@Roboneet can you take a look at this PR? I'm not sure what parts of it are out of date.

**Comment by amellnik** [comment]
This looks good to me, thanks @Roboneet !

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ julia> x = rand(10)
 
 julia> @code_js identity(x)
 let model = (function () {
-  let math = dl.ENV.math;
+  let math = tf;
```
Comment (MikeInnes) [inline README.md:26]: Since this isn't saving us any characters any more, let's just use `tf` directly everywhere :)

**Comment by MikeInnes** [comment]
Thanks a lot @Roboneet!

**Pull request merged**